build/images/ovs/apply-patches.sh

```diff
@@ -35,48 +35,50 @@ function version_let() { test "$(printf '%s\n' "$@" | sort -V | head -n 1)" == "
-
-# This patch (post 2.13.x) ensures that ovs-vswitchd does not delete datapath
-# ports on exit.
-curl https://github.com/openvswitch/ovs/commit/7cc77b301f80a63cd4893198d82be0eef303f731.patch | \
-    git apply
-
 # This patch (post 2.13.0) ensures that ct_nw_src/ct_nw_dst supports IP Mask.
 if version_let "$OVS_VERSION" "2.13.0"; then
     curl https://github.com/openvswitch/ovs/commit/1740aaf49dad6f533705dc3dce8d955a1840052a.patch | \
         git apply
 fi
 
-# These patches (post 2.13.x) are needed to fix the debian build on Ubuntu 20.04.
-curl https://github.com/openvswitch/ovs/commit/c101cd4171cfe04e214f858b4bbe089e56f13f9b.patch | \
-    git apply
-curl https://github.com/openvswitch/ovs/commit/3c18bb0fe9f23308061217f72e2245f0e311b20b.patch | \
-    git apply
-curl https://github.com/openvswitch/ovs/commit/fe175ac17352ceb2dbc9958112b4b1bc114d82f0.patch | \
-    git apply
+if version_get "$OVS_VERSION" "2.13.0" && version_let "$OVS_VERSION" "2.13.1" ; then
+    # These 2 patches (post 2.13.x) ensures that datapath flows are not deleted on
+    # ovs-vswitchd exit by default. Antrea relies on this to support hitless upgrade
+    # of the Agent DaemonSet.
+    # The second patch depends on the first one.
+    curl https://github.com/openvswitch/ovs/commit/586cd3101e7fda54d14fb5bf12d847f35d968627.patch | \
+        git apply
+    # We exclude 2 files which are likely to cause conflicts.
+    curl https://github.com/openvswitch/ovs/commit/79eadafeb1b47a3871cb792aa972f6e4d89d1a0b.patch | \
+        git apply --exclude NEWS --exclude vswitchd/ovs-vswitchd.8.in
+
+    # This patch (post 2.13.x) ensures that ovs-vswitchd does not delete datapath
+    # ports on exit.
+    curl https://github.com/openvswitch/ovs/commit/7cc77b301f80a63cd4893198d82be0eef303f731.patch | \
+        git apply
 
-# The OVS ovs-monitor-ipsec script has a Python3 shebang but still includes some Python2-specific code.
-# Until the patch which fixes the script is merged upstream, we apply it here, or Antrea IPsec support will be broken.
-curl https://github.com/lzhecheng/ovs/commit/869b06356e389079861962160e864df609d033e5.patch | \
-    git apply
+    # These patches (post 2.13.x) are needed to fix the debian build on Ubuntu 20.04.
+    curl https://github.com/openvswitch/ovs/commit/c101cd4171cfe04e214f858b4bbe089e56f13f9b.patch | \
+        git apply
+    curl https://github.com/openvswitch/ovs/commit/3c18bb0fe9f23308061217f72e2245f0e311b20b.patch | \
+        git apply
+    curl https://github.com/openvswitch/ovs/commit/fe175ac17352ceb2dbc9958112b4b1bc114d82f0.patch | \
+        git apply
+
+    # The OVS ovs-monitor-ipsec script has a Python3 shebang but still includes some Python2-specific code.
+    # Until the patch which fixes the script is merged upstream, we apply it here, or Antrea IPsec support will be broken.
+    curl https://github.com/lzhecheng/ovs/commit/869b06356e389079861962160e864df609d033e5.patch | \
```

Comment:
do you think you could use the commit reference from the OVS repo now that your patch was merged?

Reply:
This patch is still there. It is on a different branch from the one merged into OVS.

Reply:
I understand. I was suggesting using this link if possible: https://github.com/openvswitch/ovs/commit/8a09c2590ef2ea0edc250ec46e3d41bd5874b4ab.patch?

Reply:
You are right. Updated.